ls1intum/Artemis
# `General`: Add a feature toggle for exam live statistics

### Checklist
#### General


#### Client

### Motivation and Context



We want to toggle the exam live statistics on a general level. In order to achieve this, we added a feature toggle for this feature.

### Description


We added the `@FeatureToggle(Feature.ExamLiveStatistics)` to the corresponding methods.

### Steps for Testing

Prerequisites:
- 1 Instructor
- One/Multiple Students
- Exam

1. Log in to Artemis
2. Create an Exam
3. Register multiple students
4. Navigate to the Exam Live Statistics dashboard
5. Enable the Exam Live Statistics
6. Participate with the students in the exam
7. Verify that actions are collected
8. Disable the exam live statistics via the feature toggle
9. Verify that the exam live statistics is disabled and cannot be enabled
10. Verify that the students client does not send any actions via websocket 
12. Enable the exam live statistics
13. Verify that the old actions are deleted (reload the exam live statistics dashboard)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|

_unchanged_

### Screenshots



Disabled buttons, checkboxes and links:

<details>
<summary>Screenshots</summary>

![disabled_enable_button_on_monitoring_page](https://user-images.githubusercontent.com/60005702/181905769-ea21d17e-013f-438f-b24c-ffe3a9160ef6.png)
![disabled_in_exam_configuration](https://user-images.githubusercontent.com/60005702/181905770-c1da81d5-0c15-4aaf-9a6f-2c66a3b105d8.png)
![disabled_on_exam_checklist](https://user-images.githubusercontent.com/60005702/181905771-c078722e-9e35-4bbb-8303-0a391da90363.png)

![image](https://user-images.githubusercontent.com/60005702/182131086-75628e87-6ae0-40fe-8db1-f1eb83d8530c.png)
![image](https://user-images.githubusercontent.com/60005702/182131135-95b40126-db44-4fd5-ac8f-89842de215d7.png)

</details>
